Assembly prework

rchopra opened the pull request:
Got the basics working and attempted the stretch goals. I have no idea what I'm doing is the right away to go about this sort of thing -- was just experimenting with a bunch of stuff. Tried to leave lots of comments to explain the nonsense.

rchopra committed:
Implement basic sum_to_n

rchopra committed:
Implement binary_convert

rchopra committed:
Detect pangrams without punctuation

rchopra committed:
Filter out punctuation with conditionals

rchopra committed:
Replace convoluted steps with bts

After making a first pass through the exercise, I checked the hints and
learned about the `bts` instruction. This is a lot more direct than the
manual shift-ing and or-ing I was doing.

rchopra committed:
Remove conditionals for non-alpha checking

rchopra committed:
Add some more punctuation to tests

arlandism commented on systems/assembly/pangram.asm:
```diff
@@ -1,4 +1,25 @@
+  jne check          ; jump and keep checking if not all 1s seen
+  mov rax, 1         ; otherwise, set the return value to true and return
+done:
 	ret
```
Really impressed by the terseness of this 👏

rchopra replied:
You are kind to say this but seeing your solution I feel like it's a lot more elegant to do something like subtract 'A' than this weird `xor`ing business. 🤣

rchopra committed:
Subtract A instead of xor in pangram

This was a good idea Arlandis and Oz had in their solutions that's quite
elegant.
